### PR TITLE
Add data processing for SPM7

### PR DESCRIPTION
**Pull request by danielhuppmann**
This PR adds the basic config for importing data for SPM7 - but I'm at a bit of a loss on how to get the workflow to parse this...

The script that would yield some IAMC-compatible output is
```python
import pandas as pd
import pyam

# read the data
data = pd.read_csv("SPM7_data.csv")[0:3]

# wrange with the raw data
data = data.T
variables = data.loc["Unnamed: 0"].values
data.columns = variables
data = data[1:]  # remove first line which has variable names
data.index.name = "scenario"
data["year"] = 2100

# cast to an IamDataFrame
df = pyam.IamDataFrame(data.reset_index(), model="IAMs", region="World", value=variables, unit="PgC")

# rename the variable
mapping = dict([(i, "Cumulative Carbon Stored|" + i.split(" ")[0].title()) for i in df.variable])
df.rename(variable=mapping, inplace=True)

# export cleaned data as csv
df.to_csv(<path>)
```

The "year" time domain here is understood as "the year until which cumulative carbon is counted".

Also, in my opinion, only the results in PgC should be stored in this repo, because the values in GtCO2 and as shares can easily be calculated...

### CODE REVIEW

**Comment by danielhuppmann** [comment]
@znicholls, can you help me to fit this script into the right place and add the necessary plumbing around it?

File: compilation-config.yaml [hunk 1]
```diff
@@ -21,6 +21,15 @@ fig_wg1_spm4:
   output_data_path: data/processed/fig-spm4/fig-spm4-timeseries.csv
   compile_function: openscm_ar6_wg1_data_compilation.compile_spm_fig_4_timeseries
 
+fig_wg1_spm7:
+  variables:
+      Cumulative Carbon Stored|Atmosphere:
```
Comment (znicholls) [inline compilation-config.yaml]:
```suggestion
      Cumulative Carbon Uptake|Atmosphere:
```

I would suggest using uptake as this is the term that is more commonly used (but not a massive deal either way)

**Comment by znicholls** [comment]
> @znicholls, can you help me to fit this script into the right place and add the necessary plumbing around it?

https://github.com/danielhuppmann/AR6-WG1-Data-Compilation/pull/1, wasn't quite as obvious as I'd hoped but hopefully all makes sense now (or at least more sense)

File: compilation-config.yaml (hunk 1 above)
Comment (danielhuppmann) [reply compilation-config.yaml]: Changed this to uptake

**Comment by danielhuppmann** [comment]
Thanks for the assist, @znicholls!

File: compilation-config.yaml [hunk 2]
```diff
@@ -21,6 +21,16 @@ fig_wg1_spm4:
...
+      Cumulative Carbon Stored|Land:
```
Comment (znicholls) [inline compilation-config.yaml]: Is it just me or does the config yaml still have 'Stored' even though the output has 'Uptake'?

Comment (danielhuppmann) [reply compilation-config.yaml]: Thanks for catching that!

**Pull request merged**